packages/stems/src/components/HarmonyButton/HarmonyButton.module.css

```diff
@@ -0,0 +1,220 @@
+/* ===Base Styles=== */
+.button {
+  --button-color: var(--primary);
+  --text-color: var(--static-white);
+  --overlay-color: transparent;
+  --overlay-opacity: 0;
+  align-items: center;
+  border: 1px solid var(--button-color);
+  border-radius: var(--unit-1);
+  box-sizing: border-box;
+  color: var(--text-color);
+  cursor: pointer;
+  display: inline-flex;
+  flex-shrink: 0;
+  justify-content: center;
+  overflow: hidden;
+  position: relative;
+  text-align: center;
+  user-select: none;
+  white-space: nowrap;
+}
+
+.button:focus {
+  outline: none !important;
+}
+
+/* Only add hover styles on devices which support it */
+@media (hover: hover) {
+  .button:not(.disabled):hover {
+    transition: all var(--hover);
+    transform: scale(1.04);
+  }
+}
+
+.button:not(.disabled):active {
+  transition: all var(--press);
+  transform: scale(0.98);
+}
+
+.button.disabled {
+  pointer-events: none;
+}
+
+/* Overlay used for hover/press styling */
+.button::before {
+  content: '';
+  position: absolute;
+  display: block;
+  top: 0;
+  left: 0;
+  width: 100%;
+  height: 100%;
+  background: var(--overlay-color, transparent);
+  opacity: var(--overlay-opacity, 0);
+  pointer-events: none;
+}
+
+.icon path {
+  fill: currentColor;
+}
+
+.fullWidth {
+  width: 100%;
+}
+
+/* === Sizes === */
+
+/* Small */
+.small {
+  gap: var(--unit-1);
+  height: var(--unit-8);
+  padding: var(--unit-2) var(--unit-3);
+}
+
+.small .icon {
+  width: var(--unit-4);
+  height: var(--unit-4);
+}
+
+.small .text {
+  font-size: var(--font-s);
+  font-weight: var(--font-bold);
+  line-height: var(--unit-4);
+  text-transform: capitalize;
+}
+
+/* Default */
+.default {
+  gap: var(--unit-2);
+  height: var(--unit-12);
+  padding: var(--unit-3) var(--unit-6);
+}
+
+.default .icon {
+  width: var(--unit-5);
+  height: var(--unit-5);
+}
+
+.default .text {
+  font-size: var(--font-l);
+  font-weight: var(--font-bold);
+  line-height: calc(4.5 * var(--unit));
+  text-transform: capitalize;
+}
+
+/* Large */
+.large {
+  gap: var(--unit-2);
+  height: var(--unit-15);
+  padding: var(--unit-5) var(--unit-6);
+}
+
+.large .icon {
```

Comment:
i think this is totally good, but wondering how you feel about specificity changes here? ive personally always tried to just have classname directly tied to the element, so i might do `largeIcon` class. Im no expert though so would love your insight here

Reply:
and for example if we pass in iconClassName, doesnt specificity get a bit messed up when having to compete against `.default .icon`?

Reply:
Yeah good call. I was actually trying to use the least specificity possible, but I didn't really want to do separate classes for each piece (button, icon, text). I used a handy specificity calculator to check, and you are correct! The only way to make sure the class can be overridden is to use only one class, no descendants or combinators. Will update.